File: hanoiTowers.js
const hanoi = (height, from, to) => {
  if (height === 1) {
    console.log("Move disk 1 from peg %d to peg %d", from, to);
    return;
  }
 
  const temporary = 6 - from - to;
  hanoi(height - 1, from, temporary);
  console.log("Move disk %d from peg %d to peg %d", height, from, to);
  hanoi(height - 1, temporary, to);
};


hanoi(20, 1, 3);

File: hanoiTowers.cpp
#include <iostream>

using namespace std;

void hanoiTowers(int height, int from, int to) {
    if (height == 1) {
        cout << "Move disk 1 from peg " << from << " to peg " << to << endl;
        return;
    } 

    int temporary = 6 - from - to;
    hanoiTowers(height - 1, from, temporary);
    cout << "Move disk " << height << " from peg " << from << " to peg " << to << endl; 
    hanoiTowers(height - 1, temporary, to);

}

int main() {
    hanoiTowers(20, 1, 3);
    return 0;
}